dev-tools/iota-private-network/bootstrap.sh

```diff
@@ -108,23 +184,35 @@ create_folder_for_postgres() {
 
 main() {
   if [[ "$OSTYPE" != "darwin"* && "$EUID" -ne 0 ]]; then
-    echo "Please run as root or with sudo"
-    exit 1
-  fi
-  [ -d "$TEMP_EXPORT_DIR" ] && rm -rf "$TEMP_EXPORT_DIR"
+      echo "Please run as root or with sudo"
+      exit 1
+    fi
+    [ -d "$TEMP_EXPORT_DIR" ] && rm -rf "$TEMP_EXPORT_DIR"
 
-  [ -d "$PRIVATE_DATA_DIR" ] && ./cleanup.sh
+    [ -d "$PRIVATE_DATA_DIR" ] && ./cleanup.sh
 
-  for config_path in "$GENESIS_TEMPLATE" "$OVERLAY_PATH"; do
-    check_configs_exist "$config_path"
-  done
+    # Generate genesis template if missing
+    generate_genesis_template_if_missing
 
-  for image in "iotaledger/iota-tools" "iotaledger/iota-node" "iotaledger/iota-indexer"; do
-    check_docker_image_exist "$image"
-  done
+    # Only check overlay file existence
+    check_configs_exist "$OVERLAY_PATH"
+
+    for image in "iotaledger/iota-tools" "iotaledger/iota-node" "iotaledger/iota-indexer"; do
+      check_docker_image_exist "$image"
+    done
 
-  generate_genesis_files
-  create_folder_for_postgres
+    generate_genesis_files
+    create_folder_for_postgres
+
+  # Manage CONSENSUS_PROTOCOL in .env: set if provided, otherwise remove it
+  ENV_FILE="$PRIVNET_DIR/.env"
+  if [ -n "$PROTOCOL" ]; then
+    set_env_var CONSENSUS_PROTOCOL "$PROTOCOL" "$ENV_FILE"
+    echo "Set CONSENSUS_PROTOCOL=$PROTOCOL in $ENV_FILE"
+  else
+    unset_env_var CONSENSUS_PROTOCOL "$ENV_FILE"
+    echo "Cleared CONSENSUS_PROTOCOL from $ENV_FILE (no -p provided)"
+  fi
```

Comment:
Is it possible to export the env variable in `run.sh`?  That way there's no need to set and unset `CONSENSUS_PROTOCOL` in env file in advance. 
`./run.sh -n 10 -p starfish`